keras/src/backend/tensorflow/core.py

```diff
@@ -237,9 +237,14 @@ def map(f, xs):
     def get_fn_output_signature(x):
         out = f(x)
         return tree.map_structure(tf.TensorSpec.from_tensor, out)
-
-    fn_output_signature = get_fn_output_signature(xs[0])
-    return tf.map_fn(f, xs, fn_output_signature=fn_output_signature)
+    
+    if isinstance(xs,dict) or isinstance(xs,list) or isinstance(xs,tuple):
```

Comment:
I believe this line could be replaced with `tree.is_nested(xs)`.

Reply:
@james77777778 - Let me check with same and update here if it will works fine.